src/commands/deployWorkspaceProject/getDefaultValues/getDefaultAcrResources.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import type { Registry } from "@azure/arm-containerregistry";
+import type { ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../utils/localize";
+import { AcrListStep } from "../../deployImage/imageSource/containerRegistry/acr/AcrListStep";
+import { DeployWorkspaceProjectSettings } from "../DeployWorkspaceProjectSettings";
+
+interface DefaultAcrResources {
+    registry?: Registry;
+    imageName?: string;
+}
+
+export async function getDefaultAcrResources(context: ISubscriptionActionContext, settings: DeployWorkspaceProjectSettings | undefined): Promise<DefaultAcrResources> {
+    const noMatchingResource = { registry: undefined };
+
+    if (!settings || !settings.containerRegistryName) {
+        return noMatchingResource;
+    }
+
+    const registries: Registry[] = await AcrListStep.getRegistries(context);
+    const savedRegistry: Registry | undefined = registries.find(r => r.name === settings.containerRegistryName);
+
+    if (savedRegistry) {
+        ext.outputChannel.appendLog(localize('foundResourceMatch', 'Used saved workspace settings and found an existing container registry.'));
```

Comment:
Big fan of the logs in this function. You could include the found resource names in the log if you wanted to be even more verbose. But just an idea.

Reply:
Yeah, actually this suggestion will be added in the following PR!